spec/controllers/alchemy/admin/pages_controller_spec.rb

```diff
@@ -571,6 +571,56 @@ module Alchemy
             alchemy_get :edit, id: page.id
           end
         end
+
+        context 'if page is scoped' do
+          context 'to a single role' do
+            context 'user role does not match' do
+              before do
+                expect_any_instance_of(Page).to receive(:definition).at_least(:once) { {"editable_by" => ["freelancer"]} }
```

Comment:
You already test `editable_by?`, so it's ok to stub this method instead leaking knowledge of the model implementation here.

```ruby
allow_any_instance_of(Page).to receive(:editable_by?) { false }
```